wweval/R/ms_fig4.R

```diff
@@ -0,0 +1,519 @@
+#' Make a CRPS density plot for a subset of locations
+#'
+#' @param scores A tibble of scores by location, forecast date, date and model,
+#' the ouput of `scoringutils::score()` on samples.
+#' @param horizons_to_show A vector of strings indicating the names of the
+#' `horizon` that we want to show on the plot, must be a subset of
+#' `nowcast`, `1 wk`, `2 wks`,`3 wks`, `4 wks` and `overall`
+#'
+#' @return a ggplot object that is a vertical facet of violin plots colored
+#' by model type and broken down my horizon
+#' @export
+make_fig4_crps_density <- function(scores,
+                                   horizons_to_show = c(
+                                     "nowcast",
+                                     "1 wk", "4 wks",
+                                     "overall"
+                                   )) {
+  scores_by_horizon <- scores |>
+    data.table::as.data.table() |>
```

Comment:
As above, will this definitely be a `tibble` rather than a `data.table` when passed? If not, need to correct docs and can omit the `as.data.table()` call.